x-pack/qa/kerberos-tests/src/javaRestTest/java/org/elasticsearch/xpack/security/authc/kerberos/KerberosAuthenticationIT.java

```diff
@@ -157,8 +157,22 @@ public void testGetOauth2TokenInExchangeForKerberosTickets() throws PrivilegedAc
     @SuppressForbidden(reason = "SPNEGO relies on hostnames and we need to ensure host isn't a IP address")
     protected HttpHost buildHttpHost(String host, int port) {
         try {
-            InetAddress inetAddress = InetAddress.getByName(host);
-            return super.buildHttpHost(inetAddress.getCanonicalHostName(), port);
+            final InetAddress address = InetAddress.getByName(host);
+            final String hostname = address.getCanonicalHostName();
+            // InetAddress#getCanonicalHostName depends on the system configuration (e.g. /etc/hosts) to return the FQDN.
+            // In case InetAddress cannot resolve the FQDN it will return the textual representation of the IP address.
```

Comment:
Could this instead be due to the security manager not having permissions to delegate ?

It looks like we already added a few aliases here: 
https://github.com/elastic/elasticsearch/blob/main/x-pack/qa/kerberos-tests/src/javaRestTest/resources/plugin-security.policy via https://github.com/elastic/elasticsearch/pull/32588

I wonder if due to implementation order (order defined?) we just pick a different alias and the security manager is preventing the use of the chosen alias. If that is the case, then I think we should change the comments here and maybe look into a different fix ?

related: https://docs.oracle.com/en/java/javase/17/docs/api/java.security.jgss/javax/security/auth/kerberos/DelegationPermission.html

Reply:
Also, I am not too familiar with this test, but could we just 
```java
final InetAddress address = InetAddress.getByName(host)
String hostname = "localhost";
if (address.isLoopbackAddress() == false){
   hostname = address.getCanonicalHostName();
}
return super.buildHttpHost(hostname, port);
```
So that we use a fixed name to refer to any loopback addresses ?  (also do we even test non-loopback address?)

Reply:
> Could this instead be due to the security manager not having permissions to delegate ?

I've checked it and it's not caused by the security manager. I've [tested this](https://gist.github.com/slobodanadamovic/57057bf1741e86a95ee560475dfad2e3) with custom `hosts` file.

> do we even test non-loopback address?

We don't. We resolve only `127.0.0.1` which is defined in `build.gradle` file:

https://github.com/elastic/elasticsearch/blob/e4ff839e4cab6d443d1b43d10733029ae0afecd3/x-pack/qa/kerberos-tests/build.gradle#L19

and we expect it later to be resolved to `localhost` since we set up `HTTP/localhost` in keytab file:
https://github.com/elastic/elasticsearch/blob/cbda0a51c654b6b0657abd3b46813de712edb34d/test/fixtures/krb5kdc-fixture/src/main/resources/provision/peppa.sh#L6

> So that we use a fixed name to refer to any loopback addresses ?

Initially I took similar approach to return `localhost` without even attempting to resolve `127.0.0.1` address, but then decided to actually try to resolve it and fail if it's not resolved. Reasoning was to make it future proof and fail fast in case the `127.0.0.1` address gets changed for some reason.

Reply:
thanks for the additional explaintion